Form 22-1990 Progress Bar

cvalarida opened the pull request:
The original issue (department-of-veterans-affairs/vets.gov-team#906) was to just get rid of the subway and use the segmented progress bar. Without the subway, it doesn't make sense to have the border around the content, and it's a bit off-putting to not have the title (as shown below for the 22-1995):

![image](https://cloud.githubusercontent.com/assets/12970166/22614577/8eaeaeae-ea38-11e6-992c-1ddfe7d039d2.png)


I'm not sure what to do with the headers. Sometimes they make sense (Personal Information --> Direct deposit):

![image](https://cloud.githubusercontent.com/assets/12970166/22614510/a3b68642-ea37-11e6-84f0-fbc19c05f3f9.png)

And sometimes they're redundant (Veteran Information --> Veteran information)

![image](https://cloud.githubusercontent.com/assets/12970166/22614516/bdf8faf8-ea37-11e6-9ba2-806523242167.png)

I didn't want to over-implement without getting the green light on design, so this is what it looks like now. Suggestions?

cvalarida committed:
Clarifies params for SegmentedProgressBar

Based on what I found in common/schemaform/FormNav.jsx, both params for SegmentedProgressBar are numbers.

cvalarida committed:
Remove subway, add SegmentedProgressBar

cvalarida committed:
Limit progress bar to 8 columns

jbalboni commented on src/js/edu-benefits/1990/containers/EduBenefitsApp.jsx:
```diff
@@ -75,6 +75,14 @@ class EduBenefitsApp extends React.Component {
 
+    // There's got to be a better way to do this using schemaform, but for now...
+    // Find the index of the chapter...
+    const currentIndex = _.findIndex(chapters, (chapter) => {
```
There should be logic for this in the NavHeader component.

jbalboni commented:
@cvalarida You can remove the headers on each page. We can have @goldenmeanie go through them and see if there are any that are necessary for specific pages, but the default should be that we don't show them.

jbalboni commented on src/js/edu-benefits/1990/containers/EduBenefitsApp.jsx:
```diff
@@ -93,16 +101,16 @@ class EduBenefitsApp extends React.Component {
-              pages={pageState}
-              chapters={chapters}
-              currentUrl={currentLocation.pathname}/>
+        <div className="medium-8 columns">
```
Does this need to be in a separate row, or can it be in the same div as NavHeader?

cvalarida replied:
It can be in the same div, but when I tried it, the NavHeader and progress bar just got squashed together:

![image](https://cloud.githubusercontent.com/assets/12970166/22655753/fc66c6c0-ec45-11e6-8ab7-99830bea76c5.png)

I'll see what I can do to separate them while they're in the same div.

cvalarida replied:
Look good?

![image](https://cloud.githubusercontent.com/assets/12970166/22657491/8f0649ba-ec4b-11e6-80f2-642d2eca22fa.png)

jbalboni replied:
Looks good to me!

jbalboni commented on src/js/edu-benefits/1990/containers/EduBenefitsApp.jsx:
```diff
@@ -93,16 +101,16 @@ class EduBenefitsApp extends React.Component {
+          {
+            // Only render SegmentedProgressBar if we have a current chapter
+            // Could move this into SegmentedProgressBar
+            currentIndex ? <SegmentedProgressBar total={chapters.length} current={currentIndex}/> : null
```
Hiding this when we don't have a chapter will probably work, but the actual logic is that we don't want this to show up on the introduction and submit-message pages. You could check for that specifically, which would make what's happening clearer.

jbalboni commented:
@cvalarida It probably does make sense to include a title like the 1995 form. You can use the text from the introduction page (Apply for education benefits).

goldenmeanie commented:
Is origin/1990-progress-bar up to date. I'll pull it down and take a spin through to try to clear this up!

jbalboni commented:
@goldenmeanie Should be.

cvalarida committed:
Encapsulate progress bar and header in one div

cvalarida committed:
Borrow chapter-finding logic

Definitely not DRY, but I don't know that refactoring this bit is a high priority (or even makes sense to do right now).

cvalarida committed:
Add title to 1990

jbalboni commented:
@cvalarida It looks like there's a linting error that's causing the build check to fail above.

cvalarida committed:
Removes headers for pages

cvalarida committed:
Remove pageState (now unused)

I'm not sure where it's being passed from, but eslint was unhappy that it was still there.

cvalarida committed:
Clarify when to render progress bar

cvalarida commented:
To sum up the changes:

---
![image](https://cloud.githubusercontent.com/assets/12970166/22666883/3c7b86e6-ec6f-11e6-8f68-4900887e34fd.png)

---
![image](https://cloud.githubusercontent.com/assets/12970166/22666836/17b0ed9c-ec6f-11e6-9a9f-b004b53beceb.png)

---
![image](https://cloud.githubusercontent.com/assets/12970166/22666807/01915678-ec6f-11e6-93f3-e71e09986505.png)

cvalarida committed:
Remove hairline

cvalarida committed:
Adds some headers back in

cvalarida committed:
Match introduction alignment to title

cvalarida committed:
Pacify eslint

cvalarida commented:
Removed the border and aligned the intro page with the title to match the 22-1995.

![image](https://cloud.githubusercontent.com/assets/12970166/22676712/f521c200-eca1-11e6-97a5-3d713cbe7ef0.png)

---
While still keeping the indented forms.

![image](https://cloud.githubusercontent.com/assets/12970166/22676727/0dfe2764-eca2-11e6-9965-cfd548e88dc7.png)

jbalboni reviewed (approved):
Made a couple comments, but this looks good. One other thing I noticed from trying the form locally, you can remove the "Review Application" header on the review page, too.

cvalarida committed:
Code cleanup using classNames and &&

jbalboni commented:
@cvalarida Looks good to me. I think you can merge this once you fix the merge conflict (and the build passes).

cvalarida committed:
Pacifying Jenkins' linter

cvalarida committed:
More pacifying...

It'd be nice to have the same linting as Jenkins. It's being worked on. In the meantime...

cvalarida committed:
Merge branch 'master' into 1990-progress-bar
